src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1020,7 +1020,7 @@ namespace cryptonote
     for (auto& tx : arg.txs)
     {
       tx_verification_context tvc{};
-      if (!m_core.handle_incoming_tx({tx, crypto::null_hash}, tvc, tx_relay, true))
+      if (!m_core.handle_incoming_tx({tx, crypto::null_hash}, tvc, tx_relay, true) && !tvc.m_no_drop_offense)
```

Comment:
It looks like a node would relay a tx that has `m_no_drop_offense` set to true because it continues executing past the if statement.

Looks like you'd want something more like:

```cpp
if (!m_core.handle_incoming_tx({tx, crypto::null_hash}, tvc, tx_relay, true))
{
        if (tvc.m_no_drop_offense)
        {          
            LOG_PRINT_CCONTEXT_L1("Tx verification failed with a no drop offense");
            continue;
        }
        LOG_PRINT_CCONTEXT_L1("Tx verification failed, dropping connection");
        drop_connection(context, false, false);
        return 1;
}
```

Reply:
Nevermind, `tvc.m_relay` doesn't get set so should remain set to defaulted `relay_method::none` so it should fall into the final case of the switch. Sorry

Reply:
I've added a debug log to check this on my node, and it's never anything else but `relay_method::none`